src/main/java/sirius/biz/packages/Packages.java

```diff
@@ -0,0 +1,119 @@
+     */
+    public String getPackageName(String scope, String packageKey) {
+        return NLS.get("Packages.package." + scope + "." + packageKey);
+    }
+
+    /**
+     * Returns the list of upgrades available for the given scope.
+     *
+     * @param scope the scope
+     * @return the upgrades of the scope
+     */
+    public List<String> getUpgrades(String scope) {
+        return getScopeExtension(scope).getStringList("upgrades");
+    }
+
+    /**
+     * Returns the translated name for the given upgrade of the given scope.
+     *
+     * @param scope      the scope
+     * @param upgradeKey the upgrade key
+     * @return the translated name for the given upgrade of the given scope
+     */
+    public String getUpgradeName(String scope, String upgradeKey) {
+        return NLS.get("Packages.upgrade." + scope + "." + upgradeKey);
+    }
+
+    /**
+     * Checks whether the current user has the required permission for a given permission.
+     *
+     * @param permission the permission in question
+     * @return true if the current user has the required permission, false if not
+     */
+    public boolean hasRequiredPermissionForPermission(String permission) {
+        UserInfo currentUser = UserContext.get().getUser();
+        return hasRequiredPermissionForPermission(permission, currentUser::hasPermission);
+    }
+
+    /**
+     * Checks whether the permissionData object has the required permission for a given permission.
+     *
+     * @param permission     the permission in question
+     * @param permissionData the permissionData object to check
+     * @return true if the permissionData object has the required permission, false if not
+     */
+    public boolean hasRequiredPermissionForPermission(String permission, PermissionData permissionData) {
+        return hasRequiredPermissionForPermission(permission,
+                                                  requiredPermission -> permissionData.getPermissions()
+                                                                                      .contains(requiredPermission));
+    }
+
+    /**
+     * Checks whether some object has the required permission for a given permission, via a predicate.
+     * <p>
+     * The predicate will be called with the required permission for the permission and should return true or false,
+     * depending on if the required permission is present.
+     *
+     * @param permission             the permission in question
+     * @param hasPermissionPredicate a predicate which will determine if the required permission is present
+     * @return true if the object has the required permission (determined via the predicate), false if not
+     */
+    public boolean hasRequiredPermissionForPermission(String permission, Predicate<String> hasPermissionPredicate) {
+        Config config = Sirius.getSettings().getConfig("security.packages.required-permissions-for-permission");
```

Comment:
maybe move the hasPermission logic of UserInfo to here (as separate method) and also apply it here)